sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/PartitionPumpManager.java

```diff
@@ -366,11 +367,12 @@ private void cleanup(PartitionOwnership claimedOwnership, PartitionPump partitio
      */
     private Context startProcessTracingSpan(EventData eventData, String eventHubName, String fullyQualifiedNamespace) {
         Object diagnosticId = eventData.getProperties().get(DIAGNOSTIC_ID_KEY);
-        if (diagnosticId == null || !tracerProvider.isEnabled()) {
+        if (tracerProvider == null || !tracerProvider.isEnabled()) {
             return Context.NONE;
         }
 
-        Context spanContext = tracerProvider.extractContext(diagnosticId.toString(), Context.NONE)
+        Context spanContext = Objects.isNull(diagnosticId) ? Context.NONE : tracerProvider.extractContext(diagnosticId.toString(), Context.NONE);
+        spanContext = spanContext
```

Comment:
Will this `diagnosticId.toString()` result in an NPE when diagnosticId is null?

Reply:
> Will this `diagnosticId.toString()` result in an NPE when diagnosticId is null?

the former code check diagnosticId before call the toStirng() method. The new code use Objects.IsNull() to check diagnosticId. BTW this PR is closed. And here is the new opened PR https://github.com/Azure/azure-sdk-for-java/pull/26180